File: packages/agent/extensions/explore/ast/languages/fixtures/sample.java
package com.demo;

import java.util.List;

/** Greeter type. */
public class Greeter {
    private int count;
    public Greeter() {}
    public void hello() {}
    public static final int MAX = 10;
}

interface Named {
    void name();
}

enum Color {
    RED,
    GREEN;
    void paint() {}
}

File: packages/agent/src/ast/languages/fixtures/sample.go
package demo

import (
	"fmt"
	"io"
)

// Greeter greets callers.
type Greeter struct {
	Name string
	count int
}

// Hello prints a greeting.
func (g *Greeter) Hello() {
	fmt.Println("hi", g.Name)
}

type Writer interface {
	io.Writer
	Flush() error
}

type Color int

const (
	Red Color = iota
	Green
)

var DefaultName = "world"

func NewGreeter(name string) *Greeter {
	return &Greeter{Name: name}
}

func hidden() {}
